test/integration/places_api_test.rb

```diff
@@ -327,6 +343,23 @@ class PlacesAPITest < ActionDispatch::IntegrationTest
           assert_equal @place3.name, data["places"][1]["name"]
         end
       end
+
+      context "when the service is missing a SNAC" do
+        setup do
+          @service.update(local_authority_hierarchy_match_type: Service::LOCAL_AUTHORITY_COUNTY_MATCH)
+        end
+
+        should "only return the county results in order of nearness, not the district ones for postcodes in a county+district council hierarchy" do
+          stub_locations_api_has_location("EX39 1QS", [{ "latitude" => 51.05318361810428, "longitude" => -4.191071523498792, "local_custodian_code" => 1234 }])
+          stub_local_links_manager_has_a_local_authority("my-county", snac: nil, gss: "E060000063", local_custodian_code: 1234)
```

Comment:
Unless I'm misunderstanding from looking at this[ line](https://github.com/alphagov/local-links-manager/pull/1080/commits/849855c4d8370e8fa6eaeb3c7147acd8c35addd9#diff-b1eac0dc5137f1b4fda6084fa074b5cb5656ba23eb3a8718329933696226f7d1R15) if the snac was nil, wouldn't the snac key-value pair be omitted from the hash rather than having a value of nil?

Reply:
That's right - but here we're using a stub helper method, and these parameters go into making up one of those hashes. The test helper method defaults to giving you a SNAC, so in the call to this parameter we have to explicitly set it to nil so that it won't appear in the faked API result this test sees.